### DISCUSSION TITLE
Give multiple users access to a workspace

### DISCUSSION THREAD

**Discussion by mkyrilov** [eclipse-theia/theia-cloud#337] [Q&A]
Hello everyone,

I am busy integrating `@eclipse-theiacloud/common` into my own application, and I am using the following API to create a workspace:

```js
TheiaCloud.Workspace.createWorkspace(
  {
    appId: 'APP_ID',
    appDefinition: 'APP_DEFINITION',
    user: 'USER-A@EXAMPLE.COM',
    label: 'LABEL',
    serviceUrl: 'SERVICE_URL',
  },
  { accessToken: 'ACCESS_TOKEN' },
);
```

This operation is successful and I am able to launch the workspace by calling the following:

```js
TheiaCloud.Session.startSession(
  {
     appId: 'APP_ID',
     user: 'USER-A@EXAMPLE.COM',
     appDefinition: 'APP_DEFINITION',
     workspaceName: 'WORKSPACE_NAME', // obtained from the createWorkspace operation above
     serviceUrl: 'SERVICE_URL',
  },
  { accessToken: 'ACCESS_TOKEN' },
);
```

In the above, `USER-A@EXAMPLE.COM` is able to launch their workspace and everything works fine.

My question is, how would I specify that `USER-B@EXAMPLE.COM` should also have access to that same workspace?

**Comment by jfaltermeier** [comment]
Hi,

Currently, our focus has been on workspaces that are owned by users, and as such, we do not support sharing at the moment.

Regarding your use case, could you clarify whether you expect both users to be able to work on the workspace simultaneously? 
Or are you envisioning a scenario where a workspace can be shared, but only one user can actively use it at any given time?

**Comment by mkyrilov** [comment]
Hello, we are envisioning something where more than one user can access the workspace simultaneously. It is basically for a team lead to be able to access the workspaces of all their team members, but not necessarily collaborate with them in real time.

We were able to make this work by modifying the `*-emailconfig-*` configmap which gets created along with the session. All we had to do was add more email addresses, and the proxy lets those users in. The problem is that the configmap is not a persisted object, as it disappears when the session is destroyed.

If it was possible to specify a list of users when creating a workspace, and the operator to write that list in the configmap, then this would fulfill our use case.
